src/lightgbm_R.cpp

```diff
@@ -449,15 +449,23 @@ LGBM_SE LGBM_BoosterGetEvalNames_R(LGBM_SE handle,
   R_API_BEGIN();
   int len;
   CHECK_CALL(LGBM_BoosterGetEvalCounts(R_GET_PTR(handle), &len));
+
+  const size_t reserved_string_size = 128;
   std::vector<std::vector<char>> names(len);
   std::vector<char*> ptr_names(len);
   for (int i = 0; i < len; ++i) {
-    names[i].resize(128);
+    names[i].resize(reserved_string_size);
     ptr_names[i] = names[i].data();
   }
+
   int out_len;
-  CHECK_CALL(LGBM_BoosterGetEvalNames(R_GET_PTR(handle), &out_len, ptr_names.data()));
+  size_t required_string_size;
+  CHECK_CALL(LGBM_BoosterGetEvalNames(R_GET_PTR(handle),
+                                      len, &out_len,
+                                      reserved_string_size, &required_string_size,
+                                      ptr_names.data()));
```

Comment:
Looks ok to me. Could you change the indentation to make it clearer?

```
CHECK_CALL(
    LGBM_BoosterGetEvalNames(
        R_GET_PTR(handle),
        len,
        &out_len,
        reserved_string_size,
        &required_string_size,
        ptr_names.data()
    )
);
```

Reply:
Sure :)

Reply:
Thanks @jameslamb :)